### DISCUSSION TITLE
Req for debian release or instructions for build/compile on ubuntu will help.

### DISCUSSION THREAD

**Discussion by lalamax3d** [AcademySoftwareFoundation/OpenRV#81] [Ideas]
thanks for sharing amazing tool.

**Comment by rogernelson** [comment]
Hi @lalamax3d.  Although we only officially support CentOS as a Linux distro for the time being, a few users have told us that they successfully built OpenRV on Ubuntu.  I suggest following the CentOS build guide and replacing the commands with Ubuntu equivalents (`apt-get` instead of `yum`, etc).  If you run into any problems, feel free to post here and we'll do our best to guide you through them.

**Reply by rubikstriangle** [reply]
I gave this a try on ubuntu 22.04  The first issue I ran into is that devtoolset-9 isn't available for Ninja, I tried converting the rpm to deb with ninja but installing that didn't install scl.  I suspect that there are a train of dependancies that need to be installed as well.  Any thoughts on working around that?
I skipped that and tried:
**cmake -B cmake-build -G Ninja -DRV_DEPS_QT5_LOCATION=$HOME/Qt/5.15.2/gcc_64 -DCMAKE_BUILD_TYPE=Release**
but this failed with the following message:
-- Looking for CL_VERSION_3_0
-- Looking for CL_VERSION_3_0 - not found
-- Looking for CL_VERSION_2_2
-- Looking for CL_VERSION_2_2 - not found
-- Looking for CL_VERSION_2_1
-- Looking for CL_VERSION_2_1 - not found
-- Looking for CL_VERSION_2_0
-- Looking for CL_VERSION_2_0 - not found
-- Looking for CL_VERSION_1_2
-- Looking for CL_VERSION_1_2 - not found
-- Looking for CL_VERSION_1_1
-- Looking for CL_VERSION_1_1 - not found
-- Looking for CL_VERSION_1_0
-- Looking for CL_VERSION_1_0 - not found
CMake Error at /usr/local/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find OpenCL (missing: OpenCL_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/local/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/share/cmake-3.24/Modules/FindOpenCL.cmake:163 (find_package_handle_standard_args)
  src/lib/ip/IPCore/CMakeLists.txt:255 (FIND_PACKAGE)


I have previously compiled and built OpenRV on centos 7 and copied a release version to Ubuntu and it works just fine.  I also needed to copy these two libraries from centos7 /lib64/libva-drm.so.1 and /lib64/libva.so.1 to the rv lib folder.  Seems like this works great for Ubuntu but I thought I'd post my findings with Ubuntu just in case there is any helpful input.

**Reply by aditiapratama** [reply]
I can pass this using `sudo apt install ocl-icd-opencl-dev`

**Comment by rogernelson** [comment]
Hi @rubikstriangle.  As you notice, you don't need devtoolset per se.  The reason this is in the build guide is that the default gcc toolchain (gcc, g++, ld, etc) that comes with CentOS is too old to compile RV.  If you don't run into any compilation errors, the one you have installed already is probably up-to-date enough.  For reference here are the versions that come with devtoolset 9 if you want to compare:
https://access.redhat.com/documentation/en-us/red_hat_developer_toolset/9/html/9.0_release_notes/dts9.0_release

The errors you see are because OpenCL doesn't seem to be installed.  If you search google for OpenCL and your version of Ubuntu, you'll probably find out how to get it.

For libva, you can probably just install that with `sudo apt-get install libva1`.  One handy tip, is on your CentOS machine, if you want to know which package a particular file comes from you can run `rpm -qf [path to file]` and it will tell you which package installed that file.  From there, you can probably figure out what the package name is on Ubuntu and the install that with `sudo apt-get`.  

Hope that helps.

**Comment by aditiapratama** [comment]
Hello, is `rvbootstrap` command not supported for ubuntu?

I got this following error on Pop!Os

```
CMake Error at cmake/defaults/rv_targets.cmake:116 (MESSAGE):
  Unknown or unsupported Linux distribution version; stopping configuration!
Call Stack (most recent call first):
  CMakeLists.txt:88 (INCLUDE)
 ```

**Reply by aditiapratama** [reply]
ok disable this line run the setup
```
# IF(RHEL_VERBOSE)
  #   MESSAGE(STATUS "Redhat Entreprise Linux version: ${RHEL_VERBOSE}")
  #   STRING(REGEX MATCH "([0-9]+)" RHEL_VERSION_MAJOR "${RHEL_VERBOSE}")
  #   SET(RV_TARGET_IS_RHEL${RHEL_VERSION_MAJOR}
  #       BOOL TRUE "Detected a Redhat Entreprise Linux OS"
  #   )
  # ELSE()
  #   MESSAGE(FATAL_ERROR "Unknown or unsupported Linux distribution version; stopping configuration!")
  # ENDIF()
  
 ```

**Reply by geffrak** [reply]
The build is only supported on RHEL.

**Comment by akdbra** [comment]
Hey Everyone,
I was able to get to this point on Debian 11.

```
-- Configuring done
CMake Error at src/lib/app/OutputVideoDevices/CMakeLists.txt:36 (TARGET_LINK_LIBRARIES):
  Target "OutputVideoDevices" links to:

    OpenGL::GLX

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at src/lib/graphics/TwkGLFFBO/CMakeLists.txt:44 (TARGET_LINK_LIBRARIES):
  Target "TwkGLFFBO" links to:

    OpenGL::GLX

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.


```

Details during bootstrap - 
```
-- Using Boost:        1.76.0
-- Using Dav1d:        1.0.0
-- Using FFMPEG:       n4.4.3
-- Using GC:           8.2.2
-- Using GLEW:         2.2.0
-- Using Imath:        3.1.5
-- Using OpenEXR:      3.1.7
-- Using Python3:      3.9.15
-- Using PySide2:      5.15.2.1
-- Using Qt5:          5.15.2
-- Using zlib:         1.2.13
```
cmake version 3.24.0
ninja version 1.10.1

Is there any idea what can be done to get further?

**Reply by geffrak** [reply]
The reason is that we don't support building with anything else than RHEL

**Reply by akdbra** [reply]
Is there no plan to support Debian/Ubuntu builds in the future?

**Reply by geffrak** [reply]
It's not impossible, but it's not in the plans at the moment to build with something other than RHEL. If it's something of value, you can voice it to the Open Review Initiative of the ASWF slack channel or in one of the OpenRV monthly meetings.
